packages/widget-angular/README.md

```diff
@@ -70,15 +70,96 @@ export class AppComponent {
-| visibility        | 'maximized' \| 'minimized' \| 'hidden' |
+| Prop                     | Type                                   |
+| ------------------------ | -------------------------------------- |
+| license                  | string (required)                      |
+| customerName             | string                                 |
+| group                    | string                                 |
+| customerEmail            | string                                 |
+| chatBetweenGroups        | boolean                                |
+| sessionVariables         | Record<string, string>                 |
+| visibility               | 'maximized' \| 'minimized' \| 'hidden' |
+| customIdentityProvider\* | () => {                                |
+|                          | getToken: () => Promise<string>        |
+|                          | getFreshToken: () => Promise<string>   |
+|                          | hasToken: () => Promise<boolean>       |
+|                          | invalidate: () => Promise<void>        |
+|                          | }                                      |
+
+#### Custom Identity Provider
+
+In order to make Custom Identity Provider work, you'll have to properly implement and provide a set of following methods:
+
+- `getToken` - resolving Chat Widget token. If you want to cache the token, this should return the cached token instead of a fresh request to https://accounts.livechat.com/customer/token endpoint.
+- `getFreshToken` - resolving Chat Widget token. This should always make a call for a fresh token from https://accounts.livechat.com/customer/token endpoint.
+- `hasToken` - resolving boolean. It determines whether a token has been acquired.
+- `invalidate` - resolving nothing. When called, it should remove the current token. There is no need to do anything else as a new token will be requested by getFreshToken afterward.
+
+##### Example usage
+
+```ts
+// app.component.ts
+
+import { Component } from '@angular/core'
+import { EventHandlerPayload } from '@livechat/widget-angular'
+
+@Component({
+  /* ... */
+  templateUrl: './app.component.html',
+})
+export class AppComponent {
+  customIdentityProvider() {
+    const baseAPI = 'YOUR_API_URL'
+    const userId = '30317220-c72d-11ed-2137-0242ac120002'
+
+    const getToken = async () => {
+      const response = await fetch(`${baseAPI}/getToken/${userId}`)
+
+      const token = await response.json()
+      console.log('getToken', token)
+      return token
+    }
+
+    const getFreshToken = async () => {
+      const response = await fetch(`${baseAPI}/getFreshToken/${userId}`)
+
+      const token = await response.json()
+      console.log('getFreshToken, token')
+      return token
+    }
+
+    const hasToken = async () => {
+      const response = await fetch(`${baseAPI}/hasToken/${userId}`)
+      const data = await response.json()
+      return JSON.stringify(data) === 'true'
```

Comment:
```suggestion
      return data
```
Why use the `JSON.stringify` and not just return the data ? 🤔

Reply:
Yep, we can simplify that. I thought about being more explicit that it has to return `Boolean`.